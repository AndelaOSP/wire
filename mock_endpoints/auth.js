const auth = (req, res, next) => {
  const token = req.query.token || req.headers['authorization'];
  if (token == 'null' || token == 'false') {
    res.status(401).send({
      error: true,
      message: 'Sorry. You might not be authorized.',
      status: 'error'
    });
  } else {
    return next();
  }
};

module.exports = auth;
