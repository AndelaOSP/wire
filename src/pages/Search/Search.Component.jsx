import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import TextField from 'material-ui/TextField';
import PropTypes from 'prop-types';

// styling
import './Search.scss';

// actions
import { searchIncidents } from '../../actions/incidentAction';

// Components
import IncidentCard from '../../Components/IncidentList/IncidentCard.Component';

/**
 * @class SearchComponent
 */
class SearchComponent extends Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.searchInput.focus();
  }

  /**
   * Method to handle search input change
   */
  handleInputChange = () => {
    let searchQuery = this.searchInput.input.value.toLowerCase();
    if (searchQuery) {
      this.props.searchIncidents(searchQuery);
    }
  };

  /**
   * Method to exit search
   * @param {event} event - Event triggering exit to dashboard
   */
  handleExit = event => {
    event.preventDefault();
    this.props.history.push('/dashboard');
  };

  getTime = timestamp => new Date(timestamp).toLocaleTimeString();

  getDate = timestamp =>
    new Date(timestamp).toLocaleDateString('en-GB', {
      day: 'numeric',
      month: 'short',
      year: 'numeric'
    });

  render() {
    const incidents = this.searchInput ? this.props.incidents : [];
    return (
      <div className="search-container">
        <TextField
          ref={input => (this.searchInput = input)}
          floatingLabelText="Search for an incident"
          fullWidth
          rows={2}
          className="input-style"
          onChange={this.handleInputChange}
        />
        <i className="fa fa-times-circle" title="Click to exit search" onClick={this.handleExit} />
        <div className="incident-cards">
          {incidents.length
            ? incidents.map(incident => (
                <IncidentCard
                  key={incident.id}
                  incidentId={incident.id}
                  incidentSubject={incident.subject}
                  incidentDescription={incident.description || 'No Description'}
                  incidentReportDate={`reported on ${this.getDate(incident.dateOccurred)}`}
                  incidentTime={this.getTime(incident.dateOccurred)}
                  incidentFlag={incident.Level.name}
                  assignees={incident.assignees}
                />
              ))
            : null}
        </div>
      </div>
    );
  }
}

/**
 * Search Component Props validation
 */
SearchComponent.propTypes = {
  history: PropTypes.object.isRequired,
  incidents: PropTypes.array.isRequired,
  searchIncidents: PropTypes.func.isRequired,
};

/**
 * map state from the store to props
 * @param {*} state
 * @returns {*} partial state
 */
const mapStateToProps = state => {
  return {
    incidents: state.incidents,
  };
};

/**
 * map dispatch to props
 * @param {*} dispatch
 */
const mapDispatchToProps = dispatch =>
  bindActionCreators(
    {
      searchIncidents,
    },
    dispatch
  );

export default connect(mapStateToProps, mapDispatchToProps, null, { withRef: true })(SearchComponent);
