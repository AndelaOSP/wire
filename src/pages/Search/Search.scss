@import '../../theme.scss';

.search-container {
    width: 70vw;
    margin-top: 10vh;
    margin-right: auto;
    margin-left: auto;
    color: '#000';
}

.fa-times-circle:hover {
    color: map_get($red, 300);
}

.fa-times-circle {
    float: right;
    position: absolute;
    margin-top: 2.5vh;
    color: map_get($grey, 200);
    font-size: 1.7rem !important;
    cursor: pointer !important;    
}

.input-style {
    &>input {
        box-sizing: border-box !important;
        font-size: 3rem !important;
        font-weight: 200 !important;
        margin-top: 0.7rem !important;
        color: '#929292'!important;
        cursor: text !important;

        &:focus {
            &+div hr {
                border-bottom: 1px solid map_get($blue, 300) !important;
            }
        }
    }
}
