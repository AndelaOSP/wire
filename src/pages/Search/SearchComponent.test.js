import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import { SearchComponent, mapDispatchToProps, mapStateToProps } from './Search.Component';
import { testIncidents } from '../../../mock_endpoints/mockData';

describe('SearchComponent', () => {
  describe('Component rendering and methods', () => {
    let wrapper;
    let shallow;
    const props = {
      classes: {
        input: {},
        focused: {},
      },
      history: {
        push: jest.fn(),
      },
      incidents: testIncidents,
      searchIncidents: jest.fn(),
    };

    beforeEach(() => {
      shallow = createShallow();
      wrapper = shallow(<SearchComponent {...props} />);
    });

    it('should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should should change the searchQuery state when handleInputChange is called', () => {
      wrapper.find('TextField').simulate('change', {
        target: { name: 'searchQuery', value: 'subject' },
      });

      expect(wrapper.instance().state.searchQuery).toEqual('subject');
    });

    it('should call the history push method when times button is clicked', () => {
      wrapper.find('.fa-times-circle').simulate('click', {
        preventDefault: jest.fn(),
      });

      expect(props.history.push).toHaveBeenCalled();
    });
  });

  describe('The mapStateToProps', () => {
    it('should return the expected props', () => {
      const state = {
        incidents: testIncidents,
      };
      const props = mapStateToProps(state);

      expect(props.incidents).toEqual(state.incidents);
    });
  });

  describe('The mapDispatchToProps', () => {
    it('should dispatch actions', () => {
      const dispatch = jest.fn(() => Promise.resolve());
      const props = mapDispatchToProps(dispatch);

      props.searchIncidents();
      expect(dispatch).toHaveBeenCalled();
    });
  });
});
