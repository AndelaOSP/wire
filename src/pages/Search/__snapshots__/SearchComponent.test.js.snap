// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchComponent Component rendering and methods should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchComponent
    classes={
      Object {
        "focused": Object {},
        "input": Object {},
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    incidents={
      Array [
        Object {
          "Level": Object {
            "name": "Red",
          },
          "Location": Object {
            "centre": "Nairobi",
            "country": "Kenya",
            "name": "Quiet Room",
          },
          "Status": Object {
            "status": "In Progress",
          },
          "assignees": Array [
            Object {
              "assignedRole": "ccd",
              "createdAt": "2018-03-20T15:45:48.778Z",
              "email": "caroline.nkirote@andela.com",
              "id": "-KhLsLqucbC1WAMylcFt",
              "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
              "roleId": 2,
              "updatedAt": "2018-03-20T15:45:48.778Z",
              "username": "Caroline Nkirote",
            },
            Object {
              "assignedRole": "assignee",
              "createdAt": "2018-03-20T15:45:48.778Z",
              "email": "mercy.muchai@andela.com",
              "id": "-KhLsOxrKcKZC8i2n888",
              "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
              "roleId": 2,
              "updatedAt": "2018-03-20T15:45:48.778Z",
              "username": "Mercy Muchai",
            },
          ],
          "categoryId": 4,
          "chats": Array [],
          "createdAt": "2018-03-20T15:48:15.380Z",
          "dateOccurred": "2018-02-02T00:00:00.000Z",
          "description": "Some random thing happened and I didnt like it",
          "id": "cjezu2kr700010wx1yy6kxu5z",
          "levelId": 1,
          "locationId": "cjezu2kqo00000wx1a5u0i2xt",
          "notes": Array [],
          "reporter": Object {
            "createdAt": "2018-03-20T15:45:48.778Z",
            "email": "caroline.nkirote@andela.com",
            "id": "-KhLsLqucbC1WAMylcFt",
            "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
            "roleId": 2,
            "updatedAt": "2018-03-20T15:45:48.778Z",
            "username": "Caroline Nkirote",
          },
          "statusId": 2,
          "subject": "Theft",
          "updatedAt": "2018-03-21T09:02:41.309Z",
        },
        Object {
          "Level": Object {
            "name": "Yellow",
          },
          "Location": Object {
            "centre": "Lagos",
            "country": "Nigeria",
            "name": "First Floor",
          },
          "Status": Object {
            "status": "In Progress",
          },
          "assignees": Array [
            Object {
              "assignedRole": "ccd",
              "createdAt": "2018-03-20T15:45:48.778Z",
              "email": "caroline.nkirote@andela.com",
              "id": "-KhLsLqucbC1WAMylcFt",
              "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
              "roleId": 2,
              "updatedAt": "2018-03-20T15:45:48.778Z",
              "username": "Caroline Nkirote",
            },
            Object {
              "assignedRole": "assignee",
              "createdAt": "2018-03-20T15:45:48.778Z",
              "email": "mercy.muchai@andela.com",
              "id": "-KhLsOxrKcKZC8i2n888",
              "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
              "roleId": 2,
              "updatedAt": "2018-03-20T15:45:48.778Z",
              "username": "Mercy Muchai",
            },
          ],
          "categoryId": 4,
          "createdAt": "2018-03-20T15:48:15.380Z",
          "dateOccurred": "2018-02-02T00:00:00.000Z",
          "description": "Some other random thing happened and I hated it",
          "id": "cjezu2kr700010wx1yy6kxu5y",
          "levelId": 1,
          "locationId": "cjezu2kqo00000wx1a5u0i2xt",
          "reporter": Object {
            "createdAt": "2018-03-20T15:45:48.778Z",
            "email": "caroline.nkirote@andela.com",
            "id": "-KhLsLqucbC1WAMylcFt",
            "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
            "roleId": 2,
            "updatedAt": "2018-03-20T15:45:48.778Z",
            "username": "Caroline Nkirote",
          },
          "statusId": 2,
          "subject": "Fighting",
          "updatedAt": "2018-03-21T09:02:41.309Z",
        },
      ]
    }
    searchIncidents={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TextField
          InputProps={
            Object {
              "classes": Object {
                "focused": Object {},
                "root": Object {},
              },
            }
          }
          autoFocus={true}
          fullWidth={true}
          label="Search for an incident"
          name="title"
          onChange={[Function]}
          required={false}
          rows={2}
          select={false}
          value=""
          variant="standard"
        />,
        <i
          className="fa fa-times-circle"
          onClick={[Function]}
          title="Click to exit search"
        />,
        <div
          className="incident-cards"
        >
          <IncidentCard
            assignees={
              Array [
                Object {
                  "assignedRole": "ccd",
                  "createdAt": "2018-03-20T15:45:48.778Z",
                  "email": "caroline.nkirote@andela.com",
                  "id": "-KhLsLqucbC1WAMylcFt",
                  "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
                  "roleId": 2,
                  "updatedAt": "2018-03-20T15:45:48.778Z",
                  "username": "Caroline Nkirote",
                },
                Object {
                  "assignedRole": "assignee",
                  "createdAt": "2018-03-20T15:45:48.778Z",
                  "email": "mercy.muchai@andela.com",
                  "id": "-KhLsOxrKcKZC8i2n888",
                  "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
                  "roleId": 2,
                  "updatedAt": "2018-03-20T15:45:48.778Z",
                  "username": "Mercy Muchai",
                },
              ]
            }
            incidentDescription="Some random thing happened and I didnt like it"
            incidentFlag="Red"
            incidentId="cjezu2kr700010wx1yy6kxu5z"
            incidentReportDate="reported on 2 Feb 2018"
            incidentSubject="Theft"
            incidentTime="3:00:00 AM"
          />
          <IncidentCard
            assignees={
              Array [
                Object {
                  "assignedRole": "ccd",
                  "createdAt": "2018-03-20T15:45:48.778Z",
                  "email": "caroline.nkirote@andela.com",
                  "id": "-KhLsLqucbC1WAMylcFt",
                  "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
                  "roleId": 2,
                  "updatedAt": "2018-03-20T15:45:48.778Z",
                  "username": "Caroline Nkirote",
                },
                Object {
                  "assignedRole": "assignee",
                  "createdAt": "2018-03-20T15:45:48.778Z",
                  "email": "mercy.muchai@andela.com",
                  "id": "-KhLsOxrKcKZC8i2n888",
                  "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
                  "roleId": 2,
                  "updatedAt": "2018-03-20T15:45:48.778Z",
                  "username": "Mercy Muchai",
                },
              ]
            }
            incidentDescription="Some other random thing happened and I hated it"
            incidentFlag="Yellow"
            incidentId="cjezu2kr700010wx1yy6kxu5y"
            incidentReportDate="reported on 2 Feb 2018"
            incidentSubject="Fighting"
            incidentTime="3:00:00 AM"
          />
        </div>,
      ],
      "className": "search-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "InputProps": Object {
            "classes": Object {
              "focused": Object {},
              "root": Object {},
            },
          },
          "autoFocus": true,
          "fullWidth": true,
          "label": "Search for an incident",
          "name": "title",
          "onChange": [Function],
          "required": false,
          "rows": 2,
          "select": false,
          "value": "",
          "variant": "standard",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fa fa-times-circle",
          "onClick": [Function],
          "title": "Click to exit search",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <IncidentCard
              assignees={
                Array [
                  Object {
                    "assignedRole": "ccd",
                    "createdAt": "2018-03-20T15:45:48.778Z",
                    "email": "caroline.nkirote@andela.com",
                    "id": "-KhLsLqucbC1WAMylcFt",
                    "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
                    "roleId": 2,
                    "updatedAt": "2018-03-20T15:45:48.778Z",
                    "username": "Caroline Nkirote",
                  },
                  Object {
                    "assignedRole": "assignee",
                    "createdAt": "2018-03-20T15:45:48.778Z",
                    "email": "mercy.muchai@andela.com",
                    "id": "-KhLsOxrKcKZC8i2n888",
                    "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
                    "roleId": 2,
                    "updatedAt": "2018-03-20T15:45:48.778Z",
                    "username": "Mercy Muchai",
                  },
                ]
              }
              incidentDescription="Some random thing happened and I didnt like it"
              incidentFlag="Red"
              incidentId="cjezu2kr700010wx1yy6kxu5z"
              incidentReportDate="reported on 2 Feb 2018"
              incidentSubject="Theft"
              incidentTime="3:00:00 AM"
            />,
            <IncidentCard
              assignees={
                Array [
                  Object {
                    "assignedRole": "ccd",
                    "createdAt": "2018-03-20T15:45:48.778Z",
                    "email": "caroline.nkirote@andela.com",
                    "id": "-KhLsLqucbC1WAMylcFt",
                    "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
                    "roleId": 2,
                    "updatedAt": "2018-03-20T15:45:48.778Z",
                    "username": "Caroline Nkirote",
                  },
                  Object {
                    "assignedRole": "assignee",
                    "createdAt": "2018-03-20T15:45:48.778Z",
                    "email": "mercy.muchai@andela.com",
                    "id": "-KhLsOxrKcKZC8i2n888",
                    "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
                    "roleId": 2,
                    "updatedAt": "2018-03-20T15:45:48.778Z",
                    "username": "Mercy Muchai",
                  },
                ]
              }
              incidentDescription="Some other random thing happened and I hated it"
              incidentFlag="Yellow"
              incidentId="cjezu2kr700010wx1yy6kxu5y"
              incidentReportDate="reported on 2 Feb 2018"
              incidentSubject="Fighting"
              incidentTime="3:00:00 AM"
            />,
          ],
          "className": "incident-cards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "cjezu2kr700010wx1yy6kxu5z",
            "nodeType": "class",
            "props": Object {
              "assignees": Array [
                Object {
                  "assignedRole": "ccd",
                  "createdAt": "2018-03-20T15:45:48.778Z",
                  "email": "caroline.nkirote@andela.com",
                  "id": "-KhLsLqucbC1WAMylcFt",
                  "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
                  "roleId": 2,
                  "updatedAt": "2018-03-20T15:45:48.778Z",
                  "username": "Caroline Nkirote",
                },
                Object {
                  "assignedRole": "assignee",
                  "createdAt": "2018-03-20T15:45:48.778Z",
                  "email": "mercy.muchai@andela.com",
                  "id": "-KhLsOxrKcKZC8i2n888",
                  "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
                  "roleId": 2,
                  "updatedAt": "2018-03-20T15:45:48.778Z",
                  "username": "Mercy Muchai",
                },
              ],
              "incidentDescription": "Some random thing happened and I didnt like it",
              "incidentFlag": "Red",
              "incidentId": "cjezu2kr700010wx1yy6kxu5z",
              "incidentReportDate": "reported on 2 Feb 2018",
              "incidentSubject": "Theft",
              "incidentTime": "3:00:00 AM",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "cjezu2kr700010wx1yy6kxu5y",
            "nodeType": "class",
            "props": Object {
              "assignees": Array [
                Object {
                  "assignedRole": "ccd",
                  "createdAt": "2018-03-20T15:45:48.778Z",
                  "email": "caroline.nkirote@andela.com",
                  "id": "-KhLsLqucbC1WAMylcFt",
                  "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
                  "roleId": 2,
                  "updatedAt": "2018-03-20T15:45:48.778Z",
                  "username": "Caroline Nkirote",
                },
                Object {
                  "assignedRole": "assignee",
                  "createdAt": "2018-03-20T15:45:48.778Z",
                  "email": "mercy.muchai@andela.com",
                  "id": "-KhLsOxrKcKZC8i2n888",
                  "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
                  "roleId": 2,
                  "updatedAt": "2018-03-20T15:45:48.778Z",
                  "username": "Mercy Muchai",
                },
              ],
              "incidentDescription": "Some other random thing happened and I hated it",
              "incidentFlag": "Yellow",
              "incidentId": "cjezu2kr700010wx1yy6kxu5y",
              "incidentReportDate": "reported on 2 Feb 2018",
              "incidentSubject": "Fighting",
              "incidentTime": "3:00:00 AM",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TextField
            InputProps={
              Object {
                "classes": Object {
                  "focused": Object {},
                  "root": Object {},
                },
              }
            }
            autoFocus={true}
            fullWidth={true}
            label="Search for an incident"
            name="title"
            onChange={[Function]}
            required={false}
            rows={2}
            select={false}
            value=""
            variant="standard"
          />,
          <i
            className="fa fa-times-circle"
            onClick={[Function]}
            title="Click to exit search"
          />,
          <div
            className="incident-cards"
          >
            <IncidentCard
              assignees={
                Array [
                  Object {
                    "assignedRole": "ccd",
                    "createdAt": "2018-03-20T15:45:48.778Z",
                    "email": "caroline.nkirote@andela.com",
                    "id": "-KhLsLqucbC1WAMylcFt",
                    "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
                    "roleId": 2,
                    "updatedAt": "2018-03-20T15:45:48.778Z",
                    "username": "Caroline Nkirote",
                  },
                  Object {
                    "assignedRole": "assignee",
                    "createdAt": "2018-03-20T15:45:48.778Z",
                    "email": "mercy.muchai@andela.com",
                    "id": "-KhLsOxrKcKZC8i2n888",
                    "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
                    "roleId": 2,
                    "updatedAt": "2018-03-20T15:45:48.778Z",
                    "username": "Mercy Muchai",
                  },
                ]
              }
              incidentDescription="Some random thing happened and I didnt like it"
              incidentFlag="Red"
              incidentId="cjezu2kr700010wx1yy6kxu5z"
              incidentReportDate="reported on 2 Feb 2018"
              incidentSubject="Theft"
              incidentTime="3:00:00 AM"
            />
            <IncidentCard
              assignees={
                Array [
                  Object {
                    "assignedRole": "ccd",
                    "createdAt": "2018-03-20T15:45:48.778Z",
                    "email": "caroline.nkirote@andela.com",
                    "id": "-KhLsLqucbC1WAMylcFt",
                    "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
                    "roleId": 2,
                    "updatedAt": "2018-03-20T15:45:48.778Z",
                    "username": "Caroline Nkirote",
                  },
                  Object {
                    "assignedRole": "assignee",
                    "createdAt": "2018-03-20T15:45:48.778Z",
                    "email": "mercy.muchai@andela.com",
                    "id": "-KhLsOxrKcKZC8i2n888",
                    "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
                    "roleId": 2,
                    "updatedAt": "2018-03-20T15:45:48.778Z",
                    "username": "Mercy Muchai",
                  },
                ]
              }
              incidentDescription="Some other random thing happened and I hated it"
              incidentFlag="Yellow"
              incidentId="cjezu2kr700010wx1yy6kxu5y"
              incidentReportDate="reported on 2 Feb 2018"
              incidentSubject="Fighting"
              incidentTime="3:00:00 AM"
            />
          </div>,
        ],
        "className": "search-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "InputProps": Object {
              "classes": Object {
                "focused": Object {},
                "root": Object {},
              },
            },
            "autoFocus": true,
            "fullWidth": true,
            "label": "Search for an incident",
            "name": "title",
            "onChange": [Function],
            "required": false,
            "rows": 2,
            "select": false,
            "value": "",
            "variant": "standard",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-times-circle",
            "onClick": [Function],
            "title": "Click to exit search",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <IncidentCard
                assignees={
                  Array [
                    Object {
                      "assignedRole": "ccd",
                      "createdAt": "2018-03-20T15:45:48.778Z",
                      "email": "caroline.nkirote@andela.com",
                      "id": "-KhLsLqucbC1WAMylcFt",
                      "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
                      "roleId": 2,
                      "updatedAt": "2018-03-20T15:45:48.778Z",
                      "username": "Caroline Nkirote",
                    },
                    Object {
                      "assignedRole": "assignee",
                      "createdAt": "2018-03-20T15:45:48.778Z",
                      "email": "mercy.muchai@andela.com",
                      "id": "-KhLsOxrKcKZC8i2n888",
                      "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
                      "roleId": 2,
                      "updatedAt": "2018-03-20T15:45:48.778Z",
                      "username": "Mercy Muchai",
                    },
                  ]
                }
                incidentDescription="Some random thing happened and I didnt like it"
                incidentFlag="Red"
                incidentId="cjezu2kr700010wx1yy6kxu5z"
                incidentReportDate="reported on 2 Feb 2018"
                incidentSubject="Theft"
                incidentTime="3:00:00 AM"
              />,
              <IncidentCard
                assignees={
                  Array [
                    Object {
                      "assignedRole": "ccd",
                      "createdAt": "2018-03-20T15:45:48.778Z",
                      "email": "caroline.nkirote@andela.com",
                      "id": "-KhLsLqucbC1WAMylcFt",
                      "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
                      "roleId": 2,
                      "updatedAt": "2018-03-20T15:45:48.778Z",
                      "username": "Caroline Nkirote",
                    },
                    Object {
                      "assignedRole": "assignee",
                      "createdAt": "2018-03-20T15:45:48.778Z",
                      "email": "mercy.muchai@andela.com",
                      "id": "-KhLsOxrKcKZC8i2n888",
                      "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
                      "roleId": 2,
                      "updatedAt": "2018-03-20T15:45:48.778Z",
                      "username": "Mercy Muchai",
                    },
                  ]
                }
                incidentDescription="Some other random thing happened and I hated it"
                incidentFlag="Yellow"
                incidentId="cjezu2kr700010wx1yy6kxu5y"
                incidentReportDate="reported on 2 Feb 2018"
                incidentSubject="Fighting"
                incidentTime="3:00:00 AM"
              />,
            ],
            "className": "incident-cards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "cjezu2kr700010wx1yy6kxu5z",
              "nodeType": "class",
              "props": Object {
                "assignees": Array [
                  Object {
                    "assignedRole": "ccd",
                    "createdAt": "2018-03-20T15:45:48.778Z",
                    "email": "caroline.nkirote@andela.com",
                    "id": "-KhLsLqucbC1WAMylcFt",
                    "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
                    "roleId": 2,
                    "updatedAt": "2018-03-20T15:45:48.778Z",
                    "username": "Caroline Nkirote",
                  },
                  Object {
                    "assignedRole": "assignee",
                    "createdAt": "2018-03-20T15:45:48.778Z",
                    "email": "mercy.muchai@andela.com",
                    "id": "-KhLsOxrKcKZC8i2n888",
                    "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
                    "roleId": 2,
                    "updatedAt": "2018-03-20T15:45:48.778Z",
                    "username": "Mercy Muchai",
                  },
                ],
                "incidentDescription": "Some random thing happened and I didnt like it",
                "incidentFlag": "Red",
                "incidentId": "cjezu2kr700010wx1yy6kxu5z",
                "incidentReportDate": "reported on 2 Feb 2018",
                "incidentSubject": "Theft",
                "incidentTime": "3:00:00 AM",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "cjezu2kr700010wx1yy6kxu5y",
              "nodeType": "class",
              "props": Object {
                "assignees": Array [
                  Object {
                    "assignedRole": "ccd",
                    "createdAt": "2018-03-20T15:45:48.778Z",
                    "email": "caroline.nkirote@andela.com",
                    "id": "-KhLsLqucbC1WAMylcFt",
                    "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
                    "roleId": 2,
                    "updatedAt": "2018-03-20T15:45:48.778Z",
                    "username": "Caroline Nkirote",
                  },
                  Object {
                    "assignedRole": "assignee",
                    "createdAt": "2018-03-20T15:45:48.778Z",
                    "email": "mercy.muchai@andela.com",
                    "id": "-KhLsOxrKcKZC8i2n888",
                    "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
                    "roleId": 2,
                    "updatedAt": "2018-03-20T15:45:48.778Z",
                    "username": "Mercy Muchai",
                  },
                ],
                "incidentDescription": "Some other random thing happened and I hated it",
                "incidentFlag": "Yellow",
                "incidentId": "cjezu2kr700010wx1yy6kxu5y",
                "incidentReportDate": "reported on 2 Feb 2018",
                "incidentSubject": "Fighting",
                "incidentTime": "3:00:00 AM",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
}
`;
